pandas/core/internals/blocks.py

```diff
@@ -2112,10 +2136,28 @@ def get_values(self, dtype: DtypeObj | None = None) -> np.ndarray:
     def values_for_json(self) -> np.ndarray:
         return self.values
 
+    @cache_readonly
+    def is_numeric(self) -> bool:  # type: ignore[override]
+        dtype = self.values.dtype
+        kind = dtype.kind
+
+        return kind in "fciub"
+
+    @cache_readonly
+    def is_object(self) -> bool:  # type: ignore[override]
+        return self.values.dtype.kind == "O"
```

Comment:
are there dtypes other than object-dtype that have kind=="O"?

Reply:
Nope. FWIW, the letter codes are defined as an enum here: https://github.com/numpy/numpy/blob/f45f692abac62265b760c0a249e8b417707c47d1/numpy/core/include/numpy/ndarraytypes.h#L94-L140